backend/app/assets/javascripts/spree/backend/routes.js

```diff
@@ -1,27 +1,37 @@
-Spree.routes.checkouts_api = Spree.pathFor('api/checkouts')
-Spree.routes.classifications_api = Spree.pathFor('api/classifications')
-Spree.routes.option_value_search = Spree.pathFor('api/option_values')
-Spree.routes.option_type_search = Spree.pathFor('api/option_types')
-Spree.routes.orders_api = Spree.pathFor('api/orders')
-Spree.routes.product_search = Spree.pathFor('api/products')
-Spree.routes.admin_product_search = Spree.pathFor('admin/search/products')
-Spree.routes.shipments_api = Spree.pathFor('api/shipments')
-Spree.routes.checkouts_api = Spree.pathFor('api/checkouts')
-Spree.routes.stock_locations_api = Spree.pathFor('api/stock_locations')
-Spree.routes.taxon_products_api = Spree.pathFor('api/taxons/products')
-Spree.routes.taxons_search = Spree.pathFor('api/taxons')
-Spree.routes.user_search = Spree.pathFor('admin/search/users')
-Spree.routes.variants_api = Spree.pathFor('api/variants')
-Spree.routes.users_api = Spree.pathFor('api/users')
+/**
+ * @deprecated Spree.routes will be removed in a future release. Please use Spree.pathFor instead.
+ * See: https://github.com/solidusio/solidus/issues/3405
+ */
 
-Spree.routes.line_items_api = function(order_id) {
-  return Spree.pathFor('api/orders/' + order_id + '/line_items')
-}
+Spree.routes = new Proxy({
+  states_search: Spree.pathFor('api/states'),
+  apply_coupon_code: function(order_id) {
+    return Spree.pathFor("api/orders/" + order_id + "/coupon_codes");
+  },
+  classifications_api: Spree.pathFor('api/classifications'),
+  option_value_search: Spree.pathFor('api/option_values'),
+  option_type_search: Spree.pathFor('api/option_types'),
+  orders_api: Spree.pathFor('api/orders'),
+  product_search: Spree.pathFor('api/products'),
+  admin_product_search: Spree.pathFor('admin/search/products'),
+  shipments_api: Spree.pathFor('api/shipments'),
+  checkouts_api: Spree.pathFor('api/checkouts'),
+  stock_locations_api: Spree.pathFor('api/stock_locations'),
+  taxon_products_api: Spree.pathFor('api/taxons/products'),
+  taxons_search: Spree.pathFor('api/taxons'),
+  user_search: Spree.pathFor('admin/search/users'),
+  variants_api: Spree.pathFor('api/variants'),
+  users_api: Spree.pathFor('api/users'),
 
-Spree.routes.payments_api = function(order_id) {
-  return Spree.pathFor('api/orders/' + order_id + '/payments')
-}
+  line_items_api: function(order_id) {
+    return Spree.pathFor('api/orders/' + order_id + '/line_items')
+  },
 
-Spree.routes.stock_items_api = function(stock_location_id) {
-  return Spree.pathFor('api/stock_locations/' + stock_location_id + '/stock_items')
-}
+  payments_api: function(order_id) {
+    return Spree.pathFor('api/orders/' + order_id + '/payments')
+  },
+
+  stock_items_api: function(stock_location_id) {
+    return Spree.pathFor('api/stock_locations/' + stock_location_id + '/stock_items')
+  }
+},routes_deprecation_proxy);
```

Comment:
I think we should use `}, Spree.routes_deprecation_proxy);`, not sure how `routes_deprecation_proxy` can be visible here.

Reply:
👍